packages/synapse-client/src/SynapseClient.ts

```diff
@@ -0,0 +1,182 @@
+  WebhookServicesApi,
+  WikiPageServices2Api,
+  WikiPageServicesApi,
+} from './generated'
+import { fetchWithExponentialTimeout } from './util/fetchWithExponentialTimeout'
+
+const DEFAULT_CONFIG = new Configuration({
+  fetchApi: fetchWithExponentialTimeout,
+})
+
+/**
+ * Creates one class that encapsulates all sets of Synapse API services.
+ */
+export class SynapseClient extends BaseAPI {
+  public accessApprovalServicesClient: AccessApprovalServicesApi
+  public accessRequirementServicesClient: AccessRequirementServicesApi
+  public activityServicesClient: ActivityServicesApi
+  public agentChatServicesClient: AgentChatServicesApi
+  public asynchronousJobServicesClient: AsynchronousJobServicesApi
+  public authenticationServices2FAClient: AuthenticationServices2FAApi
+  public authenticationServicesClient: AuthenticationServicesApi
+  public certifiedUserServicesClient: CertifiedUserServicesApi
+  public challengeServicesClient: ChallengeServicesApi
+  public dataAccessServicesClient: DataAccessServicesApi
+  public discussionServicesClient: DiscussionServicesApi
+  public dockerAuthorizationServicesClient: DockerAuthorizationServicesApi
+  public dockerCommitServicesClient: DockerCommitServicesApi
+  public dockerRegistryEventServicesClient: DockerRegistryEventServicesApi
+  public doiServicesClient: DOIServicesApi
+  public downloadListServicesClient: DownloadListServicesApi
+  public drsServicesClient: DrsServicesApi
+  public entityBundleServicesV2Client: EntityBundleServicesV2Api
+  public entityServicesClient: EntityServicesApi
+  public evaluationServicesClient: EvaluationServicesApi
+  public fileServicesClient: FileServicesApi
+  public formServicesClient: FormServicesApi
+  public jsonSchemaServicesClient: JSONSchemaServicesApi
+  public logServiceClient: LogServiceApi
+  public membershipInvitationServicesClient: MembershipInvitationServicesApi
+  public membershipRequestServicesClient: MembershipRequestServicesApi
+  public messageServicesClient: MessageServicesApi
+  public openIDConnectServicesClient: OpenIDConnectServicesApi
+  public principalServicesClient: PrincipalServicesApi
+  public projectSettingsServicesClient: ProjectSettingsServicesApi
+  public recycleBinServicesClient: RecycleBinServicesApi
+  public searchServicesClient: SearchServicesApi
+  public statisticsServicesClient: StatisticsServicesApi
+  public storageReportServicesClient: StorageReportServicesApi
+  public subscriptionServicesClient: SubscriptionServicesApi
+  public tableServicesClient: TableServicesApi
+  public teamServicesClient: TeamServicesApi
+  public userGroupServicesClient: UserGroupServicesApi
+  public userProfileServicesClient: UserProfileServicesApi
+  public verificationServicesClient: VerificationServicesApi
+  public webhookServicesClient: WebhookServicesApi
+  public wikiPageServices2Client: WikiPageServices2Api
+  public wikiPageServicesClient: WikiPageServicesApi
+
+  constructor(protected configuration = DEFAULT_CONFIG) {
```

Comment:
It was a little weird to me to have the `protected` keyword in the constructor args, but this is how the parent class does it